### DISCUSSION TITLE
MaltMaterial filters

### DISCUSSION THREAD

**Discussion by Kolupsy** [bnpr/Malt#327] [Feedback & Feature Requests]
Do MaltMaterials have an attribute that makes then belong to a certain type of material, eg 'mesh', 'screen', 'light' ect? I think this would be great to have for filtering between the types. Does this make sense or would that not be a good thing to have. 
Maybe it doesnt even need a 'material type' per se but the UI elements do the filtering just from context 🤔 

![image](https://user-images.githubusercontent.com/88381426/173312659-65872237-bc34-4196-a56c-e5018c0fa3f6.png)

**Comment by pragma37** [comment]
Material properties generated from pipeline parameters have an extension type:
https://github.com/bnpr/Malt/blob/4c2143963109f7a16be7b6c672a3a51c6072e700/BlenderMalt/MaltProperties.py#L36

But the poll method must be part of the pointer property itself, so I don't think object materials could be polled?
Maybe it can be overridden somehow?

**Reply by Kolupsy** [reply]
I am currently testing out a few things. Other than node tree filtering, a mic side effect is that you can set up default nodes for those trees if you know what kind of graph type they are

**Reply by Kolupsy** [reply]
Here is what I am experimenting with https://github.com/Kolupsy/Malt/commit/a276a0473957a94cdcf4eb7f802d8070a53fef8b

**Reply by pragma37** [reply]
That would be a nice feature, but I don't think hardcoding NPR Pipeline specific stuff into BlenderMalt is the way to go.
I would rather allow PipelineGraphs to declare a host-specific default node tree, just like GraphParameters already do:
https://github.com/bnpr/Malt/blob/5868662c0e9299baac5b89c724b55d5fd2acec4d/Malt/Pipelines/NPR_Pipeline/NPR_Pipeline.py#L92

**Reply by Kolupsy** [reply]
> I don't think hardcoding NPR Pipeline specific stuff into BlenderMalt is the way to go.

I am so used to the NPR Pipeline being the default I dont even really know what is part of the NPR Pipeline in malt to be honest 😅 
I will look into GraphParameters

**Reply by pragma37** [reply]
Yeah, I can see how it can be confusing.
As a rule of thumb, anything that is not declared in BlenderMalt itself or in Malt/Pipeline.py is not guaranteed to be there.
